### PR TITLE
Added documentation for the `glob` match type.

### LINKED ISSUE #157: Document the usage of glob
I could not find any documentation of the use of glob patterns for selecting benchmarks. Luckily there was [a question](https://github.com/bos/criterion/issues/101) asked before that helped me. Is it possible to include this information somewhere?

Also including an example in the documentation of [`MatchType`](http://hackage.haskell.org/package/criterion-1.1.4.0/docs/Criterion-Main-Options.html#t:MatchType) might also help.

### PR DESCRIPTION
**Pull request by capitanbatata**
Fixes #157.

### CODE REVIEW

File: www/tutorial.md [hunk 1]
```diff
@@ -470,7 +470,10 @@ currently four ways to configure `--match`:
 * `--match prefix`: Check if the given string is a prefix of a benchmark
   path. For instance, `"foo"` will match `"foobar"`.
 
-* `--match glob`: Use the given string as a Unix-style glob pattern.
+* `--match glob`: Use the given string as a Unix-style glob pattern. Bear in
+  mind that performing a glob match on benchmarks names is done as if they were
+  file paths, so for instance both `"*/ba*"` and "*/*" will match "foo/bar",
```
Comment (RyanGlScott) [inline www/tutorial.md]: Make sure to surround all of these strings in backticks (for consistency with the rest of the tutorial).

File: Criterion/Main/Options.hs [hunk 2]
```diff
@@ -50,7 +50,11 @@ data MatchType = Prefix
                  -- ^ Match by prefix. For example, a prefix of
                  -- @\"foo\"@ will match @\"foobar\"@.
                | Glob
-                 -- ^ Match by Unix-style glob pattern.
+                 -- ^ Match by Unix-style glob pattern. When using this match
+                 -- type benchmark names are treated as if they were
+                 -- file-paths. For example, the glob patterns @"*/ba*"@ and
```
Comment (RyanGlScott) [inline Criterion/Main/Options.hs]: In Haddock, double quotes need to be escaped with a backslash (e.g., `@\"foo\"@`), or else they'll be rendered incorrectly.

File: Criterion/Main/Options.hs [hunk 3]
```diff
@@ -50,7 +50,11 @@ data MatchType = Prefix
                  -- ^ Match by prefix. For example, a prefix of
                  -- @\"foo\"@ will match @\"foobar\"@.
                | Glob
-                 -- ^ Match by Unix-style glob pattern.
+                 -- ^ Match by Unix-style glob pattern. When using this match
+                 -- type benchmark names are treated as if they were
```
Comment (RyanGlScott) [inline Criterion/Main/Options.hs]: I would use a comma to separate "type" and "benchmark" here—it reads strangely without it.

File: www/tutorial.md (hunk 1 above)
Comment (capitanbatata) [reply www/tutorial.md]: Oops, I missed that. It's fixed now.

File: Criterion/Main/Options.hs (hunk 3 above)
Comment (capitanbatata) [reply Criterion/Main/Options.hs]: Indeed. Added a comma.

File: Criterion/Main/Options.hs (hunk 2 above)
Comment (capitanbatata) [reply Criterion/Main/Options.hs]: I missed that as well :/

**Comment by RyanGlScott** [comment]
Thanks!

**Pull request merged**